### PR TITLE
add DST unitary test

### PR DESCRIPTION
**Pull request by AntoineLimouzinRTE**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

## Summary by CodeRabbit

* **Tests**
  * Added parameterized tests to broaden hour-parsing coverage, including daylight-saving-time edge cases across multiple scenarios.
  * Improved assertions to better detect inconsistencies in file validity interval parsing.
  * Centralized test helpers to drive repeated cases for consistency and clarity.

---

*Note: Internal testing improvements only; no user-facing changes.*

### CODE REVIEW

File: src/test/java/com/farao_community/farao/gridcapa/data_bridge/FileMetadataProviderTest.java [hunk 1]
```diff
@@ -233,11 +239,46 @@ void checkEmptyTimeValidityIntervalWithDailyFileAndMalformedFileName() {
...
+                                "00-FID2-701-INIT_VIRG_REFBAL_PRES_REPREVERTICES-v1.csv";
+        mockConfig("VALID_INTRADAY", "VERTICES", "HOURLY", regex, zone);
+        final Message<?> fileMessage = MessageBuilder
+                .withPayload("")
+                .setHeader(MinioAdapterConstants.DEFAULT_GRIDCAPA_FILE_NAME_METADATA_KEY, fileName)
+                .build();
+        final Map<String, String> metadataMap = new HashMap<>();
+        fileMetadataProvider.populateMetadata(fileMessage, metadataMap);
+        final String validityInterval = metadataMap.get(GRIDCAPA_FILE_VALIDITY_INTERVAL_METADATA_KEY);
+        assertNotEquals(validityInterval.split("/")[0],
```
Comment (schwitzguebel) [inline src/test/java/com/farao_community/farao/gridcapa/data_bridge/FileMetadataProviderTest.java:267]: and asset that one hour between both utc date times ?
